### PR TITLE
Feature/close button ubs pop up/#3624

### PR DESCRIPTION
**Pull request by NewArtMask**
**Achieved results:**
- added ability in 3rd step to close the popup to continue editing the order

**Preconditions**
1. The registered user is logged in (https://ita-social-projects.github.io/GreenCityClient/#/)
2. User has filled the “Деталі замовлення” and “Особисті дані” tabs

**Steps to reproduce**
1. Click on the "Далі" button, so the user is navigated to the "Підтвердження" tab
2. Click on the "Відмінити" button at the bottom of the page

**Done:** [[UBS Orders] Cannot close the pop-up to continue editing #3624](https://github.com/ita-social-projects/GreenCity/issues/3624), [[Change request to #2379, #2531, #2533, #2541, #2477, #2478...] Error messages #3625](https://github.com/ita-social-projects/GreenCity/issues/3625)

### CODE REVIEW

**Review by Yura935 — APPROVED** [review]

**Review by BobNazar — APPROVED** [review]

**Review by TarasKuzma1999 — APPROVED** [review]

**Review by slavafisher — APPROVED** [review]

**Review by SvitlanaVorobets — APPROVED** [review]

File: src/app/main/component/shared/components/form-base/form-base.component.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export class FormBaseComponent implements ComponentCanDeactivate {
             } else {
               this.router.navigate([this.previousPath]);
             }
-          } else if (isUbsOrderSubmit) {
+          } else if (isUbsOrderSubmit && confirm === null) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/shared/components/form-base/form-base.component.ts:87]: Can you use !confirm?

File: src/app/main/component/shared/components/form-base/form-base.component.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export class FormBaseComponent implements ComponentCanDeactivate {
...
             this.cancelUBSwithoutSaving();
           }
```
Comment (hnativlyubomyr) [inline src/app/main/component/shared/components/form-base/form-base.component.ts:89]: I woud refactor subscribe logic:

const currentUrl = this.router.url;
const isUBS = currentUrl.includes('ubs/order');

if (confirm) {
  this.areChangesSaved = true;
}

if (confirm && isUbsOrderSubmit) {
  this.router.navigate(['ubs', 'confirm']);
}

if (confirm && isUBS) {
  this.cancelUBSwithoutSaving();
}

if (confirm && !isUbsOrderSubmit && !isUBS) {
  this.router.navigate([this.previousPath]);
}

if (!confirm && isUbsOrderSubmit) {
  this.cancelUBSwithoutSaving();
}

File: src/app/main/component/shared/components/form-base/form-base.component.ts (hunk 1 above)
Comment (NewArtMask) [reply src/app/main/component/shared/components/form-base/form-base.component.ts:87]: Here "null" plays the role of the third parameter in confirm and must be different from "false".

File: src/app/main/component/shared/components/form-base/form-base.component.ts (hunk 2 above)
Comment (NewArtMask) [reply src/app/main/component/shared/components/form-base/form-base.component.ts:89]: Thanks. I partially changed the code, like on the refactor.

**Pull request merged**